client/relay-chain-network/src/lib.rs

```diff
@@ -0,0 +1,460 @@
+
+		RelayChainRPCClient {
+			ws_client: Arc::new(ws_client),
+			retry_strategy: ExponentialBackoff::default(),
+		}
+	}
+
+	/// Call a call to `state_call` rpc method.
+	async fn call_remote_runtime_function(
+		&self,
+		method_name: &str,
+		hash: &PHash,
+		payload: Option<Vec<u8>>,
+	) -> RelayChainResult<sp_core::Bytes> {
+		let payload_bytes = payload.map_or(sp_core::Bytes(Vec::new()), sp_core::Bytes);
+		let params = rpc_params! {
+			method_name,
+			payload_bytes,
+			hash
+		};
+		self.request("state_call", params).await
+	}
+
+	/// Subscribe to a notification stream via RPC
+	async fn subscribe<'a, R>(
+		&self,
+		sub_name: &'a str,
+		unsub_name: &'a str,
+		params: Option<ParamsSer<'a>>,
+	) -> RelayChainResult<Subscription<R>>
+	where
+		R: DeserializeOwned,
+	{
+		self.ws_client
+			.subscribe::<R>(sub_name, params, unsub_name)
+			.await
+			.map_err(|err| RelayChainError::NetworkError(sub_name.to_string(), err))
+	}
+
+	/// Perform RPC request
+	async fn request<'a, R>(
+		&self,
+		method: &'a str,
+		params: Option<ParamsSer<'a>>,
+	) -> Result<R, RelayChainError>
+	where
+		R: DeserializeOwned + std::fmt::Debug,
+	{
+		retry_notify(
+			self.retry_strategy.clone(),
+			|| async {
+				self.ws_client
+					.request(method, params.clone())
+					.await
+					.map_err(|err| backoff::Error::Transient { err, retry_after: None })
+			},
+			|e, _| {
+				tracing::warn!("Unable to complete RPC request '{}', retrying. Error {}", method, e)
+			},
+		)
+		.await
+		.map_err(|err| RelayChainError::NetworkError(method.to_string(), err))
+	}
+
+	async fn system_health(&self) -> Result<Health, RelayChainError> {
+		self.request("system_health", None).await
+	}
+
+	async fn state_get_read_proof(
+		&self,
+		storage_keys: Vec<StorageKey>,
+		at: Option<PHash>,
+	) -> Result<ReadProof<PHash>, RelayChainError> {
+		let params = rpc_params!(storage_keys, at);
+		self.request("state_getReadProof", params).await
+	}
+
+	async fn state_get_storage(
+		&self,
+		storage_key: StorageKey,
+		at: Option<PHash>,
+	) -> Result<Option<StorageData>, RelayChainError> {
+		let params = rpc_params!(storage_key, at);
+		self.request("state_getStorage", params).await
+	}
+
+	async fn chain_get_head(&self) -> Result<PHash, RelayChainError> {
+		self.request("chain_getHead", None).await
+	}
+
+	async fn chain_get_header(
+		&self,
+		hash: Option<PHash>,
+	) -> Result<Option<PHeader>, RelayChainError> {
+		let params = rpc_params!(hash);
+		self.request("chain_getHeader", params).await
+	}
+
+	async fn parachain_host_candidate_pending_availability(
+		&self,
+		at: &PHash,
+		para_id: ParaId,
+	) -> Result<Option<CommittedCandidateReceipt>, RelayChainError> {
+		let response_bytes = self
+			.call_remote_runtime_function(
+				"ParachainHost_candidate_pending_availability",
+				at,
+				Some(para_id.encode()),
+			)
+			.await?;
+
+		Ok(Option::<CommittedCandidateReceipt<PHash>>::decode(&mut &*response_bytes.0)?)
```

Comment:
```suggestion
		self
			.call_remote_runtime_function(
				"ParachainHost_candidate_pending_availability",
				at,
				Some(para_id.encode()),
			)
			.await
```

If you use my code from above.